agent/consul/state/catalog.go

```diff
@@ -369,7 +369,22 @@ func (s *Store) ensureNoNodeWithSimilarNameTxn(tx *memdb.Txn, node *structs.Node
 	for nodeIt := enodes.Next(); nodeIt != nil; nodeIt = enodes.Next() {
 		enode := nodeIt.(*structs.Node)
 		if strings.EqualFold(node.Node, enode.Node) && node.ID != enode.ID {
-			if !(enode.ID == "" && allowClashWithoutID) {
+			// Look up the existing node's Serf health check to see if it's failed.
+			// If it is, the node can be renamed.
+			enodeCheck, err := tx.First("checks", "id", enode.Node, string(structs.SerfCheckID))
+			if err != nil {
+				return fmt.Errorf("Cannot get status of node %s: %s", enode.Node, err)
+			}
+
+			var nodeHealthy bool
```

Comment:
So, if the node has no SerfHealth, you consider it safe ?
Is it the behavior you want (I think it make sense, but it would require a comment maybe)

Reply:
Yeah that's a fair point. In general nodes without Serf are fake "external nodes" like the k8s one you mentioned or also ones created by use of ESM.

I think in that case then we have to take the thing registering them as the source of truth - if it wants to change ID or overwrite a node name I guess we just have to assume that is what they intend to do.

There are edge cases where a real agent could start with same name as an existing "fake" node but I think at this point we've done out best to stop the user misconfiguring things.

I agree a comment to make that intent clear would be a good idea.